### ISSUE TITLE
[intermittent] Mozilla VPN crashes randomly

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#6755]
**Important details:**

* No clear steps to reproduce this;
* Reproduced on both Windows 10 and 11;
* Reproducibility rate once at 2 days;

**Mozilla VPN affected versions:**

* 2.15.0 (2.202304191607);
* 2.15.0 (2.202304210155);

**Platform tested:**

* Windows 10/11;

**Prerequisites:**

* Have the client opened;
* Have an account with subscription for Mozilla VPN;

**Steps to reproduce:**

1. Navigate through the app;

**Expected results:**

* There is no issues and app settings can be changed and the app can be turned on/off;

**Actual results:**

* The app crashes;

**Notes:**

* Attaching 2 logs from Win 11 and Win 10;

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-4672)
┆Reporter: Valentina Virlics

**Comment by data-sync-user** [comment]
➤ Adrienne Davenport commented:

Valentina Virlics  - Were you able to see a crash report screen when this occurred?

Also, regarding frequency, did the crashes occur once within a 48 hour period or twice within a 24 hour period?

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Nope, I did not see the crash screen after. This is tracked separately.

And, we saw it once within 48h.

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

I was able to reproduce the crash as well, while using the latest 2.15 build, on Windows 11, macOS 12.6.1 and Android 11.

On all devices I had the VPN ON for at least 1h (macOS - 1h, Windows - 3h, Android - 2h), I clicked the server from the “Select location” section from activation screen and the client crashed. 

Attaching logs from Windows 11 and macOS 12.6.1.

[^mac crash.txt]
[^win 11 crash.txt]

**Comment by data-sync-user** [comment]
➤ Naomi Kirby commented:

We are seeing a few crashes on sentry due to heap corruption, unfortunately the logs and backtraces don’t seem to be very helpful. I will see if I can try and run the VPN client under a memory analysis tool to see if that catches anything.

**Comment by data-sync-user** [comment]
➤ Naomi Kirby commented:

The next time you encounter a crash like this, can I ask you to extract the log file without re-opening the client after the crash? I am confused as to why the logs always show nothing of interest and I am wondering if the logs are getting overwritten. You can find the log files at these locations:

  * Linux: ~/.local/share/mozillavpn.txt
  * MacOS: ~/Library/Application Support/mozillavpn.txt
  * Windows: C:\Users\<name>\AppData\Roaming\mozillavpn.txt

**Comment by data-sync-user** [comment]
➤ Matt Lichtenstein commented:

Naomi Kirby is this blocked on getting logs, or are you still actively investigating? Wondering if we should demote the priority out of “blocker”

**Comment by data-sync-user** [comment]
➤ Naomi Kirby commented:

I am blocked until I can get more information about this crash. Hopefully the next time QA experiences a crash we will be able to gather more.

I have not been able to reproduce locally yet.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Naomi Kirby Unfortunately, we were not able to trigger the crash today on desktop. We will continue to try it, tomorrow. We will update this ticket with the logs you pointed out, next time we encounter the crash.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

As we did not encountered this anymore and based on team decision ( https://mozilla.slack.com/archives/C01DCUKG95E/p1683213307819809?thread_ts=1683150662.122529&cid=C01DCUKG95E ), we are marking this as Done.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Santiago Andrigo Naomi Kirby 

We encountered this while on iOS 15.7.1 device, after ~40h of having the VPN ON;

Attaching logs.

[^Mozilla VPN-2023-05-19-094001.ips]

Please let me know if we should log another ticket or reopen this one.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Update - Just experienced a crash on Windows 10, after 5 seconds from opening the app.

Attaching logs from App Data/Roaming location.

[^Windows-mozillavpn.txt]

I’ll go ahead and reopen the ticket.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Second crash today, on Windows.

Attaching logs.

[^secondcrashmozillavpn.txt]

**Comment by data-sync-user** [comment]
➤ Betty Fleming commented:

Valentina Virlics can we please create a new bug  for this, with new logs?

**Comment by data-sync-user** [comment]
➤ Naomi Kirby commented:

Valentina Virlics The tail of the log file looks iike Sentry caught the crash, were you able to submit the crash report on Windows 10 after this occurred?

**Comment by data-sync-user** [comment]
➤ Naomi Kirby commented:

Looking through the crash reports we do have, and the contents of the log, it seems like the event is suspiciously preceeded by:

[19.05.2023 01:36:20.417] Warning: qrc:/nebula/components/MZTabNavigation.qml:105:9: TypeError: Cannot read property 'x' of null (MZTabNavigation.qml:105)
[19.05.2023 01:36:21.082] (Sentry) Info: Sentry ON CRASHIt might be worth looking into the cause of that TypeError in the QML/Javascript to see if that results in anything.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Logged https://mozilla-hub.atlassian.net/browse/VPN-4840 ( https://mozilla-hub.atlassian.net/browse/VPN-4840|smart-link ) as Betty Fleming suggested.

Replied to other questions in there.
